angular/projects/global/src/lib/services/store.service.ts

```diff
@@ -0,0 +1,153 @@
+export class StoreService {
+  private currentState: any;
+
+  constructor() {
+    this.currentState = {};
+    this.state.navigationStack = [];
+  }
+
+  get state(): any {
+    return this.currentState;
+  }
+  set state(state: any) {
+    this[state.type] = state.data;
+  }
+
+  public getNavigationStack() {
+    if (this.state.navigationStack) {
+      return this.state.navigationStack;
+    } else {
+      return null;
+    }
+  }
+
+  public getLastBackUrl() {
+    const stack = this.getNavigationStack();
+    if (stack) {
+      return stack[stack.length - 1].backUrl;
+    } else {
+      return null;
+    }
```

Comment:
Similar to above.  We probably need to check that `stack[stack-length-1]` is non-null, before trying to reference `backUrl` from it.

(Optional) Same alternate pattern as above, this could be written as:
`return stack && stack[stack.length - 1] && stack[stack.length - 1].backUrl || null`